docs/troubleshooting/troubleshooting.aad.md

```diff
@@ -0,0 +1,80 @@
+# Troubleshooting: AGIC pod stuck in not ready state
+
+## Illustration
+If AGIC pod is stuck in ready state, you must be seeing the following:
+```
+$ kubectl get pods
+
+NAME                                   READY   STATUS    RESTARTS   AGE
+<AGIC-POD-NAME>                        0/1     Running   0          19s
+mic-774b9c5d7b-z4z8p                   1/1     Running   1          15m
+mic-774b9c5d7b-zrdsm                   1/1     Running   1          15m
+nmi-pv8ch       
+```
+
+## Common causes:
+1. [Stuck at creating authorizer](#agic-is-stuck-at-creating-authorizer)
+1. [Stuck getting Application Gateway](#agic-is-stuck-getting-application-gateway)
+
+## AGIC is stuck at creating authorizer
+When the AGIC pod starts, in one of the steps, AGIC tries to get an AAD (Azure Active Directory) token for the identity assigned to it. This token is then used to perform updates on the Application gateway.
+This identity can be of two types:
+    1. User Assigned Identity
+    1. Service Principal
+
+When using User Assigned identity with AGIC, AGIC has a dependency on [`AAD Pod Identity`](https://github.com/Azure/aad-pod-identity).  
+When you see your AGIC pod stuck at `Creating Authorizer` step, then the issue could be related to the setup of the user assigned identity and AAD Pod Identity.
+
+```
+$ kubectl logs <AGIC-POD-NAME>
+ERROR: logging before flag.Parse: I0628 18:09:49.947221       1 utils.go:115] Using verbosity level 3 from environment variable APPGW_VERBOSITY_LEVEL
+I0628 18:09:49.987776       1 environment.go:240] KUBERNETES_WATCHNAMESPACE is not set. Watching all available namespaces.
+I0628 18:09:49.987861       1 main.go:128] Appication Gateway Details: Subscription="xxxx" Resource Group="resgp" Name="gateway"
+I0628 18:09:49.987873       1 auth.go:46] Creating authorizer from Azure Managed Service Identity
+I0628 18:09:49.987945       1 httpserver.go:57] Starting API Server on :8123
+```
+
+`AAD Pod identity` is responsible for assigning the user assigned identity provided by the user for AGIC to the underlying AKS nodes and setup the IP table rules to allow AGIC to get AAD token for the user to assign identity from the Instance Metadata service. When you install `AAD Pod Identity` on your AKS, it will deploy two components:
```

Comment:
`AAD Pod identity` is responsible for assigning the user assigned identity provided by the user for AGIC as AGIC identity.
to in line with the "AGIC identity" is being used in the following